usecases/config/runtime/values.go

```diff
@@ -0,0 +1,108 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package runtime
+
+import (
+	"fmt"
+	"strconv"
+	"sync/atomic"
+	"time"
+
+	"gopkg.in/yaml.v3"
+)
+
+// Type represents different types that is supported in runtime configs
+type DynamicType interface {
+	int | float64 | bool | time.Duration
+}
+
+// Value represents any runtime config value. It's zero value is
+// fully usable.
+// If you want zero value with different `default`, use `NewDynamicValue` constructor.
+type DynamicValue[T DynamicType] struct {
+	// val is the dynamically chaning value.
+	val atomic.Value
+	// def represents the default value.
+	def T
+}
+
+// NewDynamicValue returns an instance of DynamicValue as passed in type
+// with passed in value as default.
+func NewDynamicValue[T DynamicType](val T) DynamicValue[T] {
+	return DynamicValue[T]{
+		def: val,
+	}
+}
+
+// Get returns a current value for the given config. It can either be dynamic value or default
+// value (if unable to get dynamic value)
+// Consumer of the dynamic config value should care only about this `Get()` api.
+func (vv *DynamicValue[T]) Get() T {
```

Comment:
why `vv` and not something simpler as `v`?

Reply:
The code evolved :). Changed to `dv` (DynamicValue)